src/Microsoft.Framework.FileSystemGlobbing/Infrastructure/MatcherContext.cs

```diff
@@ -3,275 +3,180 @@
 
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using Microsoft.Framework.FileSystemGlobbing.Abstractions;
 
 namespace Microsoft.Framework.FileSystemGlobbing.Infrastructure
 {
     public class MatcherContext
     {
+        private DirectoryInfoBase _root;
+        private IList<PatternContextBase> _includePatternContexts;
+        private IList<PatternContextBase> _excludePatternContexts;
+
         public MatcherContext(Matcher matcher, DirectoryInfoBase directoryInfo)
         {
-            Matcher = matcher;
-            DirectoryInfo = directoryInfo;
+            _root = directoryInfo;
+
+            _includePatternContexts = new List<PatternContextBase>();
             foreach (var pattern in matcher.IncludePatterns)
             {
-                if (pattern.Contains == null)
-                {
-                    IncludePatternContexts.Add(new PatternContextLinearInclude(this, pattern));
-                }
-                else
-                {
-                    IncludePatternContexts.Add(new PatternContextRaggedInclude(this, pattern));
-                }
+                _includePatternContexts.Add(pattern.Contains == null ?
+                    (PatternContextBase)new PatternContextLinearInclude(this, pattern) : new PatternContextRaggedInclude(this, pattern));
             }
+
+            _excludePatternContexts = new List<PatternContextBase>();
             foreach (var pattern in matcher.ExcludePatterns)
             {
-                if (pattern.Contains == null)
-                {
-                    ExcludePatternContexts.Add(new PatternContextLinearExclude(this, pattern));
-                }
-                else
-                {
-                    ExcludePatternContexts.Add(new PatternContextRaggedExclude(this, pattern));
-                }
+                _excludePatternContexts.Add(pattern.Contains == null ?
+                    (PatternContextBase)new PatternContextLinearExclude(this, pattern) : new PatternContextRaggedExclude(this, pattern));
```

Comment:
Yeah, it seems to be. I'll reverse the change to an `if-else` structure to gain readability.